examples/static_set/device_subsets_example.cu

```diff
@@ -0,0 +1,154 @@
+/*
+ * Copyright (c) 2023, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <cuco/static_set_ref.cuh>
+#include <cuco/storage.cuh>
+
+#include <thrust/device_vector.h>
+#include <thrust/reduce.h>
+#include <thrust/scan.h>
+
+#include <cooperative_groups.h>
+
+#include <cuda/std/array>
+
+#include <algorithm>
+#include <cstddef>
+#include <iostream>
+
+auto constexpr cg_size     = 8;   ///< A CUDA Cooperative Group of 8 threads to handle each subset
+auto constexpr window_size = 1;   ///< TODO: how to explain window size (vector length) to users
```

Comment:
memory access granularity (which may impact perfomance depending on the size of the slot type)?

Reply:
I still like referring to it as "items per thread" or "thread granularity" as it controls how many elements an individual thread processes

Reply:
Agreed, `items_per_thread` itself is definitely a less abstractive name than `window_size`.
```cpp
  cuco::storage<key_type, items_per_thread> { ... };
```
v.s.
```cpp
  cuco::aow_storage<key_type, window_size> { ... };
  // or
  cuco::window_storage<key_type, window_size> { ... };
```

Actually, the former one is not bad at all.

Reply:
The only confusion that might occur is that `items_per_thread` in e.g. CUB refers to _input_  items per thread, whilst our `items_per_thread` means _slots_ per thread. Just a minor thing. I'm ok with it. `thread_granularity` would remove the `items` part which might be less confusing but also less descriptive. Meh.